src/platform/site-wide/announcements/components/PromoBanner.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import PromoBanner from '@department-of-veterans-affairs/formation-react/PromoBanner';
+import _recordEvent from 'platform/monitoring/record-event';
+
+export default function PromoBannerWithAnalytics({
+  recordEvent = _recordEvent,
```

Comment:
Ooh, I like this. I was thinking we'd look at `window.dataLayer` to test it (:thinking: probably have to mock and unmock `global.window` for that? :thinking:), but this is much better. :100:

Reply:
Cool, I'm always happy to see someone else on board with adding DI to a function just for the sake of testing it.